subsys/suit/platform/sdfw/src/suit_plat_invoke.c

```diff
@@ -74,15 +85,53 @@ int suit_plat_invoke(suit_component_t image_handle, struct zcbor_string *invoke_
 		return SUIT_ERR_UNSUPPORTED_COMPONENT_ID;
 	}
 
+	if (suit_plat_decode_invoke_args(invoke_args, &synchronous, &timeout_ms)) {
+		LOG_ERR("suit_plat_decode_invoke_args failed");
+		return SUIT_ERR_UNSUPPORTED_PARAMETER;
+	}
+
+	/* Halt CPU to make sure that any existing app will not generate invoke events. */
+	if (synchronous) {
+		ret = suit_plat_cpu_halt(cpu_id);
```

Comment:
The changes in this file do not work.

First, if you `suit_plat_cpu_halt` here on a sychronous boot of the recovery application (first boot of the CPUAPP), there is a SecDom assert later on - I assume this is because the runaddress has not been initialized, and the underlying function actually attempts to start the processor - but I haven't analyzed this thoroughly.

However, more importantly, if the app core has already been started then restarting it again by `suit_plat_cpu_run` (which underneath simply runs `reset_mgr_init_and_boot_processor`) is by far not enough - there are many resources (such as SSF, NRFS, some peripherals) which all need to be reset separately. Some of these do not seem to be resetable in a per-domain basis (especially when it comes to NRFS).

All in all it seems that there have been some architectural decisions that resetting domains in a per core basis is not supported.

This seems the whole idea of the recovery button checked from the recovery app fall apart...

Reply:
Decided to remove the API to halt CPUs as it is impossible to implement it.